src/python/WMCore/BossAir/Plugins/SimpleCondorPlugin.py

```diff
@@ -488,197 +489,160 @@ def updateJobInformation(self, workflow, task, **kwargs):
-            ad['ExtDESIRED_Sites'] = sites
-
-            ad['CMS_JobRetryCount'] = job['retry_count']
-            ad['WMAgent_RequestName'] = job['request_name']
-
+            ad['+ExtDESIRED_Sites'] = classad.quote(str(sites))
+            ad['+CMS_JobRetryCount'] = str(job['retry_count'])
+            ad['+WMAgent_RequestName'] = classad.quote(job['request_name'])
             match = re.compile("^[a-zA-Z0-9_]+_([a-zA-Z0-9]+)-").match(job['request_name'])
             if match:
-                ad['CMSGroups'] = match.groups()[0]
+                ad['+CMSGroups'] = classad.quote(match.groups()[0])
             else:
-                ad['CMSGroups'] = classad.Value.Undefined
-
-            ad['WMAgent_JobID'] = job['jobid']
-            ad['WMAgent_SubTaskName'] = job['task_name']
-            ad['CMS_JobType'] = job['task_type']
-            ad['CMS_Type'] = activityToType(job['activity'])
-
+                ad['+CMSGroups'] = undefined
+            ad['+WMAgent_JobID'] = str(job['jobid'])
+            ad['+WMAgent_SubTaskName'] = classad.quote(job['task_name'])
+            ad['+CMS_JobType'] = classad.quote(job['task_type'])
+            ad['+CMS_Type'] = classad.quote(activityToType(job['activity']))
+     
             # Handling for AWS, cloud and opportunistic resources
-            ad['AllowOpportunistic'] = job.get('allowOpportunistic', False)
-
+            ad['+AllowOpportunistic'] = str(job.get('allowOpportunistic', False))
             if job.get('inputDataset'):
-                ad['DESIRED_CMSDataset'] = job['inputDataset']
+                ad['+DESIRED_CMSDataset'] = classad.quote(job['inputDataset'])
             else:
-                ad['DESIRED_CMSDataset'] = classad.Value.Undefined
+                ad['+DESIRED_CMSDataset'] = undefined
             if job.get('inputDatasetLocations'):
                 sites = ','.join(sorted(job['inputDatasetLocations']))
-                ad['DESIRED_CMSDataLocations'] = sites
+                ad['+DESIRED_CMSDataLocations'] = classad.quote(str(sites))
             else:
-                ad['DESIRED_CMSDataLocations'] = classad.Value.Undefined
-
+                ad['+DESIRED_CMSDataLocations'] = undefined
             if job.get('inputPileup'):
-                ad['DESIRED_CMSPileups'] = ','.join(sorted(job['inputPileup']))
+                cmsPileups=','.join(sorted(job['inputPileup']))
+                ad['+DESIRED_CMSPileups'] = classad.quote(str(cmsPileups))
             else:
-                ad['DESIRED_CMSPileups'] = classad.Value.Undefined
-
+                ad['+DESIRED_CMSPileups'] = undefined
             # HighIO and repack jobs
-            ad['Requestioslots'] = 1 if job['task_type'] in ["Merge", "Cleanup", "LogCollect"] else 0
-            ad['RequestRepackslots'] = 1 if job['task_type'] == 'Repack' else 0
-
+            ad['+Requestioslots'] = str(1 if job['task_type'] in ["Merge", "Cleanup", "LogCollect"] else 0)
```

Comment:
Kenyi, why do we pass it as a string now?

Reply:
Because the ad dictionary is used on a different class/object now. Bare classads don't really have the concept of '+', to distinguish between built-in classads like e.g.:`request_memory` and cuztomized ones like `+Requestioslots`, but the Submit class does. So, with ads with `+`  in this case, integers, booleans and classads expressions are passed as strings, and strings are passed via `classads.quote()`.

Reply:
Gotcha, thanks! It'd be worth to have this explanation in the description of the PR (first comment) for future reference ;)